Readme clarifications, minor bug fixes

ebattistella opened the pull request:
* Clarification order of length_ranges for dispersion.
* InsilicoSV version will now be written in the output VCF.
* documentation on min_intersv_dist
* clarification of contained overlap
* clarification of blacklist regions
* fix stats code that was failing when running insilicosv from inside the folder of the config file.
* fix edge case of SV placement with blacklist regions when looking to available regions after random assignment failed.
* Overlap and blacklist region types have now to be provided per input region files if multiple files.

ebattistella committed:
documentation for filter_small_chr.
Clarification order of length_ranges for dispersion
fix stats code

ebattistella committed:
fix blacklist when run out of placement possibilities

ebattistella committed:
Merge branch 'main' of github.com:PopicLab/insilicoSV into small_chr

ebattistella committed:
* test for small_chr_filter
* documenation on min_intersv_dist
* clarification of conatined overlap
* source=InsilicoSV_v{version will now be written in the output VCF header
* clarification of blacklist regions

viq854 commented on docs/use_cases.md:
```diff
@@ -153,18 +157,21 @@ variant_sets:
+* If blacklist_regions is provided in BED format, an optional fourth column (type) can be used to categorize 
+regions and filter them per variant set.
+* If blacklist_regions is provided in VCF format, the REGION_TYPE INFO field can be used instead. 
+If this field is missing, the default region type will be DEFAULT. Only the CHROM, POS, and END fields are considered from each record.
```
is DEFAULT different from "all"?

ebattistella replied:
If two blacklist files are provided and one has a region_type field, then DEFAULT regions will only be the regions from the file without a region_type field.

ebattistella committed:
add random_seed in the inputs.md

ebattistella committed:
add output adjacencies in the doc and fix an issue of symmetry causing to report non novel adjacencies

ebattistella committed:
fix VCF END and TARGET  position to respect the convention (shift by one)
END = START + LENGTH - 1
TARGET is before the insertion

ebattistella committed:
fix the blacklist region and the constraint regions, the constraints are per file.

ebattistella committed:
more fixes and debugging of the blacklist types defined per file

ebattistella committed:
Fix region names per blacklist/overlap region files.
from_beds -> from_bed for consistency with from_vcf
clean importation from vcf and bed files and normalize between blacklist/overlap regions

ebattistella committed:
Merge with main (TR operations fix)

ebattistella committed:
fix overlap ad blacklist regions with multiple region files

ebattistella committed:
Update of the documentation and add new tests with multiple overlap and blacklist files

viq854 commented on docs/use_cases.md:
```diff
@@ -193,7 +201,7 @@ There are several ways to define the relationship between an SV and a region of
 * **`"partial"`**: The constrained breakends of the SV must overlap with one of the boundaries of a selected ROI.
 * **`"containing"`**: The constrained breakends of the SV must completely contain a selected ROI.
-* **`"contained"`**: The constrained breakends of the SV must be completely contained within a selected ROI.
+* **`"contained"`**: The constrained breakends of the SV must be completely contained within a same selected ROI.
```
nit: a>the

viq854 commented on insilicosv/sv_defs.py:
```diff
@@ -216,7 +216,7 @@ def __post_init__(self):
                 f'The anchor length is smaller than the minimum overlap for a partial overlap.')
         if self.overlap_mode in [OverlapMode.TERMINAL, OverlapMode.CHROM]:
-            chk(self.roi_filter.region_kinds == ['all'], 'No overlap_region_kinds can be specified for overlap_mode'
+            chk(list(self.roi_filter.region_kinds) == ['all'], 'No overlap_region_kinds can be specified for overlap_mode '
```
nit: kinds->types/names?

viq854 reviewed (approved):
LGTM

ebattistella committed:
replace region_kind with region_types, the class feature kind by region_type
